### PR TITLE
feat(transport): pass ECN CE marks to CC

### PR DESCRIPTION
**Pull request by mxinden**
Pass ECN CE marks received through FRAME_TYPE_ACK_ECN frames to the congestion controler.

### CODE REVIEW

**Comment by mxinden** [comment]
Rebasing onto latest https://github.com/mozilla/neqo/pull/1678/ now.

**Review by larseggert — APPROVED** [review]

File: neqo-transport/src/path.rs [hunk 1]
```diff
@@ -978,8 +978,16 @@ impl Path {
         now: Instant,
     ) {
         debug_assert!(self.is_primary());
+
+        let ecn_ce_received = self.ecn_info.on_packets_acked(acked_pkts, ack_ecn);
+        if ecn_ce_received {
+            // TODO: Should congestion window reduction through CE marks result in
+            // an update to ACK delay? See on_packets_lost below.
```
Comment (mxinden) [inline neqo-transport/src/path.rs:985]: Remaining open question.

Comment (larseggert) [reply neqo-transport/src/path.rs:985]: Could you say a bit more about this? I don't see anything immediately relevant in `on_packets_lost`.

Comment (mxinden) [reply neqo-transport/src/path.rs:985]: In `Path::on_packets_lost` `if cwnd_reduced` it calls `update_ack_delay`:

https://github.com/mxinden/neqo/blob/705227436c4d9996defee13f2bbeef61a4cd29f3/neqo-transport/src/path.rs#L999-L1001

Should we do the same in `Path::on_packets_acked` if a CE marking was received and the CWND was reduced?

Comment (larseggert) [reply neqo-transport/src/path.rs:985]: I think I'm missing something. Why would we reduce cwnd when we receive ACKs? We should be reducing cwnd when we receive a CE mark on any inbound packet.

Comment (larseggert) [reply neqo-transport/src/path.rs:985]: We should treating the reception of an ECN mark identical to the detection of a loss. (At least for now.)

Comment (larseggert) [reply neqo-transport/src/path.rs:985]: I think what we're doing here in the loss case if part of our (outdated) implementation of draft-ietf-quic-ack-frequency. We should revisit what to do in the case of loss ECN as we update that implementation. See #1267.

Comment (mxinden) [reply neqo-transport/src/path.rs]: Thank you @larseggert. Updated with https://github.com/mozilla/neqo/pull/1689/commits/13fd396a565f0cce4e3d9eedd9484bf733fd4f79.

**Comment by mxinden** [comment]
@larseggert I addressed the outstanding comment. Mind taking another look?

**Review by larseggert — APPROVED** [review]

**Pull request merged**